glusterfs/server/setup.sls

```diff
@@ -58,14 +58,18 @@ glusterfs_vol_{{ name }}:
 {%- endif %}
 
 glusterfs_vol_{{ name }}_start:
+  {%- if force_compatibility %}
+  cmd.run:
+    - name: gluster volume start {{ name }}
+    - unless: gluster volume info {{ name }} | grep "Status: Started"
+    - require:
+      - cmd: glusterfs_vol_{{ name }}
+  {%- else %}
   glusterfs.started:
     - name: {{ name }}
     - require:
-      {%- if force_compatibility %}
```

Comment:
Line 61 and line 67. glusterfs.started will only be called if we are not in compatibility mode. As a result it makes no sense to keep this here. Please see line 65 and 66, only there it makes sense to still refer to cmd resource as that's what happens in compatibility mode.

Does that make sense? Maybe you could view the code as a whole and see if my reasoning makes more sense that way.